hydroflow/src/scheduled/graph.rs

```diff
@@ -409,8 +412,79 @@ impl Hydroflow {
             _phantom: PhantomData,
         }
     }
+
+    pub fn add_dependencies(&mut self, subg_id: SubgraphId, deps: DirectedEdgeSet) {
+        let base = self.subgraphs[subg_id].dependencies.current_id;
+        for (key, val) in deps.node_names {
+            self.subgraphs[subg_id]
+                .dependencies
+                .node_names
+                .insert(key + base, val);
+        }
+        for (from, to) in deps.edges {
+            self.subgraphs[subg_id]
+                .dependencies
+                .edges
+                .insert((from + base, to + base));
+        }
+        self.subgraphs[subg_id].dependencies.current_id += deps.current_id;
+    }
+
+    fn mermaid_mangle(&self, name: String, subg: usize, node: u16) -> String {
+        let res = if name.starts_with("Handoff") {
```

Comment:
not a big deal but line makes it smell like maybe the names want to be something structured like an enum rather than raw strings?

Reply:
cleaned up as per Mingwei's suggestion